### PR TITLE
Reorder `configure_model`

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Fixes https://github.com/Lightning-AI/lightning/issues/18936#issuecomment-1796405529

----
:books: Documentation preview :books:: https://pytorch-lightning--19060.org.readthedocs.build/en/19060/

cc @borda @justusschock

### CODE REVIEW

File: docs/source-pytorch/common/lightning_module.rst [hunk 1]
```diff
@@ -1065,22 +1065,19 @@ for more information.
 
 .. code-block:: python
 
+    # devices can be GPUs, TPUs, ...
+    @on_every_device
```
Comment (awaelchli) [inline docs/source-pytorch/common/lightning_module.rst:1069]: that's a little confusing 😄

File: docs/source-pytorch/common/lightning_module.rst [hunk 2]
```diff
@@ -1065,22 +1065,19 @@ for more information.
...
     def fit(self):
+        configure_callbacks()
+
         if global_rank == 0:
```
Comment (awaelchli) [inline docs/source-pytorch/common/lightning_module.rst:1073]: while you're here, could you change this to local_rank and the line below?

**Review by awaelchli — APPROVED** [review]
Can you document this in the changelog too?

File: docs/source-pytorch/common/lightning_module.rst (hunk 1 above)
Comment (carmocca) [reply docs/source-pytorch/common/lightning_module.rst:1069]: What do you suggest instead? Or do you want to remove it?

Comment (awaelchli) [reply docs/source-pytorch/common/lightning_module.rst:1069]: maybe a comment rather than a decorator?

**Review by Borda — APPROVED** [review]

**Pull request merged**